util/dvsim/Scheduler.py

```diff
@@ -98,17 +98,23 @@ def _dispatch(self, hms, old_results):
 
         while len(to_dispatch) < num_slots and self._queued:
             next_item = self._queued.pop(0)
-
             # Does next_item have any dependencies? Since we dispatch jobs by
             # target, we can assume that each of those dependencies appears
             # in old_results.
             has_failed_dep = False
             for dep in next_item.dependencies:
                 dep_status = old_results[dep]
                 assert dep_status in ['P', 'F', 'K']
-                if dep_status in ['F', 'K']:
+
+                if next_item.needs_all_dependencies_passing:
+                    if dep_status in ['F', 'K']:
+                        has_failed_dep = True
+                        break
+                else:
                     has_failed_dep = True
```

Comment:
Is this line needed in the else clause?

Reply:
Oh good point, I should have set this default parameter in line 104.